### PR TITLE
Auto grad accum example

### PR DESCRIPTION
**Pull request by muellerzr**
Showcases how to perform automatic gradient accumulation, similar to a popular feature that was toted by MosaicML awhile back.

The initial batch size starts at 256 to guarantee a CUDA OOM and the user can observe the print statement showing the new number of steps and batch size

### CODE REVIEW

File: examples/by_feature/automatic_gradient_accumulation.py [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+    num_epochs = int(config["num_epochs"])
+    seed = int(config["seed"])
+    observed_batch_size = int(config["batch_size"])
+
+    metric = evaluate.load("glue", "mrpc")
+
+    # New Code #
+    # We use the `find_executable_batch_size` decorator, passing in the desired observed batch size
+    # to train on. If a CUDA OOM error occurs, it will retry this loop cutting the batch size in
+    # half each time. From this, we can calcualte the number of gradient accumulation steps needed
```
Comment (sgugger) [inline examples/by_feature/automatic_gradient_accumulation.py:127]:
```suggestion
    # half each time. From this, we can calculate the number of gradient accumulation steps needed
```

**Review by sgugger — APPROVED** [review]
LGTM, thanks a lot!
